### ISSUE TITLE
run-pty's node process takes 100% CPU

### ISSUE THREAD

**Issue by schoettl** [lydell/run-pty#53]
A while after I start run-pty, its node process takes 99,7-100% CPU and it doesn't go down anymore until I kill *all* its jobs or I kill run-pty as a whole.

node v18.18.2
run-pty 4.1.0

But it's hard to reproduce... a while ago it was the case already a few seconds after start. Yesterday it was about 6 hours after start. When I kill and restart it it mostly works.

Any ideas how this comes or how to debug it?

I'm starting it with

```
npx run-pty run-pty.json
```

htop shows this line for run-pty where 100.6 is the CPU column:
```
  70093 jakob       20   0  680M 72308 39304 R 100.6  0.5 49:43.01 │              │  │     ├─ node /home/…/elm/node_modules/.bin/run-pty run-pty.json
```

**Comment by lydell** [comment]
Hi!

I have never noticed this myself, and I regularly run run-pty for several days.

My only guess is that the underlying node-pty library causes this. A while ago I reported an issue in node-pty for a different issue (https://github.com/lydell/run-pty/issues/45): https://github.com/microsoft/node-pty/issues/630. Some other people then commented on that issue that they had the same problem, plus high CPU usage. Maybe that’s related! Maybe you can try some other Node.js versions, or the `UV_USE_IO_URING=0` workaround!

**Comment by schoettl** [comment]
Thanks for the hint Simon! I'll just wait for updates of node or node-pty – or restart run-pty if it occurs again :)

**Comment by lydell** [comment]
I thought of a workaround that hopefully solves the high CPU problem (https://github.com/lydell/run-pty/pull/54). I released it as version `4.1.1-beta.1`. Would you like to try it, and see if the CPU problem goes away?

**Comment by schoettl** [comment]
Thanks, I now installed it with
```
npm install run-pty@4.1.1-beta.1
```

It may take two weeks until I'm confident it helped (because lately it didn't happen very regularly anymore). I'll report back.

**Comment by lydell** [comment]
Awesome!

I just realized that I just _assumed_ that you use Linux. Is that correct? Does `node -p 'process.platform'` print `linux` for you?

**Comment by schoettl** [comment]
Yes, the assumption was correct and the command prints `linux` :-)

**Comment by schoettl** [comment]
I had no issues anymore in the last three weeks with the beta version.

We could also leave the beta version and this issue open and hope for an update in node-pty.

**Comment by lydell** [comment]
Workaround released in v5.0.0.

**Comment by lydell** [comment]
Closing, see here for why: https://github.com/microsoft/node-pty/issues/630#issuecomment-2336404137

TL;DR run-pty does not take 100 % CPU anymore. The issue isn’t 100 % solved, though, but there is nothing to track in this repo about it for a while.
